src/octoprint/plugins/gcodeviewer/static/js/viewer/worker.js

```diff
@@ -480,8 +480,10 @@ var doParse = function () {
                         center_j = Number(args[j].slice(1));
                         break;
                     case "g":
-                        if (args[j].charAt(1).toLowerCase() === "2") direction = -1;
-                        if (args[j].charAt(1).toLowerCase() === "3") direction = 1;
+                        if (args[j].charAt(1) === "2" || args[j].charAt(2) === "2")
+                            direction = -1;
+                        else if (args[j].charAt(1) === "3" || args[j].charAt(2) === "3")
+                            direction = 1;
```

Comment:
That needs a further restriction to `args[j].charAt(1) === "0"` or something like `G12` could match, leading to unexpected results.

And instead of two `charAt` comparisons I'd honestly rather go with

```suggestion
                        if (args[j].charAt(1) === "2" || args[j].substring(1, 2) === "02")
                            direction = -1;
                        else if (args[j].charAt(1) === "3" || args[j].substring(1, 2) === "03")
                            direction = 1;
```

Reply:
That should be fine, as by the time it gets to here it must already be G0/G(0)1/G(0)2/G(0)3, in line 405

Reply:
Ah, I overlooked that context. Thanks for the heads-up, and I retract my statement in that case ^^